org.jacoco.core/src/org/jacoco/core/internal/instr/InstrSupport.java

```diff
@@ -227,4 +228,24 @@ public static void push(final MethodVisitor mv, final int value) {
 		}
 	}
 
+	/**
+	 * Creates {@link ClassReader}.
+	 *
+	 * @param b
+	 *            bytes of class
+	 * @return {@link ClassReader}
+	 */
+	public static ClassReader classReaderFor(final byte[] b) {
+		final byte[] originalVersion = new byte[] { b[4], b[5], b[6], b[7] };
+		if (getVersionMajor(b) == Opcodes.V12 + 1) {
```

Comment:
Add constants for this and use `NEXT_VERSION` also in other places instead of `Opcodes.V12 + 1`?

    public static final int MAX_ASM_VERSION = Opcodes.V12;
    public static final int NEXT_VERSION = MAX_ASM_VERSION + 1;

Reply:
@marchof not counting `ContentTypeDetector` all other occurrences of `V12 + 1` are in tests, where I would prefer to keep explicit `V12 + 1`.

Reply:
Sure, for tests explicit `V12 + 1` seems better. About production code this was just an idea that came in my mind to avoid "magic numbers".